### PR TITLE
vm-virtio: Implement Snapshottable trait for Pmem

### PR DESCRIPTION
**Pull request by sboeuf**
This brings the migration support to virtio-pmem device.

Signed-off-by: Sebastien Boeuf <sebastien.boeuf@intel.com>
Signed-off-by: Yi Sun <yi.y.sun@linux.intel.com>

### CODE REVIEW

File: vm-virtio/src/pmem.rs [hunk 1]
```diff
@@ -46,13 +51,33 @@ const KILL_EVENT: DeviceEventT = 1;
 // The device should be paused.
 const PAUSE_EVENT: DeviceEventT = 2;
 
-#[derive(Copy, Clone, Debug, Default)]
-#[repr(C)]
+#[derive(Copy, Clone, Debug, Default, Deserialize)]
+#[repr(C, packed)]
```
Comment (rbradford) [inline vm-virtio/src/pmem.rs:55]: Ditto here, in fact can you check the structs in the other PRs too.

Comment (rbradford) [reply vm-virtio/src/pmem.rs:55]: Kernel does not define this one as packed.

Comment (rbradford) [reply vm-virtio/src/pmem.rs:55]: Ignore me I think they are all supposed to be packed but the kernel is missing this one.

Comment (sboeuf) [reply vm-virtio/src/pmem.rs:55]: Yes they're all supposed to be packed for the configurations, that's why I added the missing ones.

**Pull request merged**